src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/HealthCertificateArray+MostRelevant.swift

```diff
@@ -117,31 +131,28 @@ extension Array where Element == HealthCertificate {
 
 		// Other Vaccination Certificate
 
-		if let otherVaccinationCertificate = sortedHealthCertificates.last(where: { $0.type == .vaccination }) {
+		if let otherVaccinationCertificate = last(where: { $0.type == .vaccination }) {
 			return otherVaccinationCertificate
 		}
 
 		// Recovery Certificate > 180 days
 
-		if let outdatedRecoveryCertificate = sortedHealthCertificates.last(where: { $0.type == .recovery }) {
+		if let outdatedRecoveryCertificate = last(where: { $0.type == .recovery }) {
 			return outdatedRecoveryCertificate
 		}
 
 		// PCR Test Certificate > 48 hours
 
-		if let outdatedPCRTestCertificate = sortedHealthCertificates.last(where: { $0.testEntry?.coronaTestType == .pcr }) {
+		if let outdatedPCRTestCertificate = last(where: { $0.testEntry?.coronaTestType == .pcr }) {
 			return outdatedPCRTestCertificate
 		}
 
 		// RAT Test Certificate > 24 hours
 
-		if let outdatedAntigenTestCertificate = sortedHealthCertificates.last(where: { $0.testEntry?.coronaTestType == .antigen }) {
+		if let outdatedAntigenTestCertificate = last(where: { $0.testEntry?.coronaTestType == .antigen }) {
 			return outdatedAntigenTestCertificate
 		}
 
-		// Fallback
-
-		return first
+		return nil
```

Comment:
fallback to nothing, is this really a good idea?

Reply:
The fall back is moved to the calling function, this way we go to the else if block, other wise the function would return the first element which is incorrect unless all the cases fail
![Screenshot 2021-08-02 at 14 14 33](https://user-images.githubusercontent.com/15270737/127860341-99448077-4de4-44a8-b4a4-20c212b91a7f.png)

Reply:
then perhaps leave a comment here. The next one is questioning the same ;)